test/allowed_lints.txt

```diff
@@ -0,0 +1,4 @@
+clippy::module_inception
+clippy::new_ret_no_self
+clippy::new_without_default
+clippy::redundant_closure
```

Comment:
I'm still not comfortable with moving this to allowed. I know https://github.com/rust-lang/rust-clippy/issues/3071 exists, but I've yet to see it flag something in our code that *can't* be fixed, even if the automatic suggestion isn't a correct fix. And for anything we decide we really prefer with an explicit closure, we can `#[allow(…)]` it.

I'm ok with moving redundant_closure to `lints_to_fix`, but I still find this lint valuable, and I imagine issue 3071 will be fixed at some point.

Reply:
Moving to lints_to_fix would be ok as well. Given that the clippy default is warn not deny, and there are known issues, it's also reasonable to just allow it and move on.